### PR TITLE
Automates announcement banner removal

### LINKED ISSUE #1610: Automate Banner Removal
- Support inclusion of a configured `End Date` for announcements, after which the banner would be filtered out of the rotation.

### PR DESCRIPTION
**Pull request by Michaeldremy**
Closes #1610

|Banner|Expires On|Age at Expiration
|---|---|---|
|Screen Shot 2022-11-09 at 5 12 25 PM| Dec. 3rd, 2022| 3 Months old|
|Screen Shot 2022-11-09 at 5 12 12 PM|Dec. 31st, 2022|8+ months old|

### CODE REVIEW

**Review by meissadia — CHANGES_REQUESTED** [review]

File: src/homepage/AnnouncementBanner.jsx [hunk 1]
```diff
@@ -140,6 +140,21 @@ export const AnnouncementBanner = ({
...
+  // Removes announcement when endDate property on an announcement is greater or equal to the currentDate
+  useEffect(() => {
+    let currentDate = new Date()
+
+    if (announcements.length >= 1) {
+      announcements.filter((item) => {
+        if (!item.props.endDate) {
+          return true
+        } else {
+          item.props.endDate >= currentDate.toLocaleDateString()
```
[the remark this answers is not in the document]
Comment (Michaeldremy) [reply src/homepage/AnnouncementBanner.jsx]: Code has been updated by removing unnecessary `useEffect` code. Only changes to this branch as of last commit are related to adding `endDate` to the configs if that banner has a `endDate`.

**Review by Michaeldremy — COMMENTED** [review]
Code has been updated to reflect code review comment.

**Review by meissadia — APPROVED** [review]
I think this gives each banner a nice long life.

**Comment by chynnakeys** [comment]
I think these dates work for me @meissadia @Michaeldremy

**Pull request merged**